### ISSUE TITLE
[Bug] CLI stops at zero length array

### ISSUE THREAD

**Issue by Benjipede** [Kipt/BEdit#47]
When I run the command line interface and an array of length zero is encountered, it stops outputting, but no error is reported like in case if a scalar is given size 0. I haven't tried the GUI.

Here is an example:

Layout file:
```
struct Thing
{
    string(1) a;
    string(1) b;
    string(1) c[a - 48]; // Just writing c[0] also produces the error.
    string(1) d;
};

layout Thing;
```
Data file:
`012`

**Comment by Kipt** [comment]
Thank you for reporting this issue. 

I can confirm that this is a command line issue only. GUI application version 0.2.0 reports error "Error: Invalid array count of 'c' (specified count is -48)".

**Comment by Kipt** [comment]
Fixed in BitBucket repository commit 0502810, will be included in next release. Support for zero-length array will probably be implemented in future releases.

I'll close this ticket when next release is made, but if you want to you can already clone the repository and build it from source.

**Comment by Kipt** [comment]
Implemented for version 0.2.1
